File: solutions-by-language/python/1304_find_n_unique_integers_sum_up_to_zero.py
class Solution:
    def sumZero(self, n: int) -> List[int]:
        nums = []
        self.fill(nums, n)
        if n % 2 :
            nums.append(0)
        return nums

    def fill(self, nums: list, n: int) :
        for i in range(1,n//2 + 1):
            nums.append(-i)
            nums.append(i)

        return

File: solutions-by-language/cpp/1304_find_n_unique_integers_sum_up_to_zero.cpp
class Solution {
public:
    vector<int> sumZero(int n) {
        vector<int> nums;
        fill(nums, n);
        if (n % 2){
            nums.push_back(0);
        }
        return nums;
    }

    void fill(vector<int>& nums, int n){
        for (int i = 1; i <= n/2; i++){
            nums.push_back(-i);
            nums.push_back(i);
        }
        return;
    }
};